File: src/org/genyris/core/Visitor.java
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.core;

import org.genyris.dl.Graph;
import org.genyris.dl.Triple;
import org.genyris.exception.GenyrisException;
import org.genyris.interp.EagerProcedure;
import org.genyris.interp.LazyProcedure;
import org.genyris.interp.TailCall;
import org.genyris.io.pipe.Pipe;
import org.genyris.java.JavaWrapper;

public interface Visitor {
	public void visitPair(Pair cons) throws GenyrisException;

	public void visitStrinG(StrinG lst) throws GenyrisException;

	public void visitLazyProc(LazyProcedure lproc) throws GenyrisException;

	public void visitEagerProc(EagerProcedure eproc) throws GenyrisException;

	public void visitDictionary(Dictionary frame) throws GenyrisException;

	public void visitBignum(Bignum bignum) throws GenyrisException;

	public void visitStandardClass(StandardClass klass) throws GenyrisException;

	public void visitExpWithEmbeddedClasses(ExpWithEmbeddedClasses exp)
			throws GenyrisException;

	public void visitSimpleSymbol(SimpleSymbol simpleSymbol)
			throws GenyrisException;

	public void visitFullyQualifiedSymbol(URISymbol sym)
			throws GenyrisException;

	public void visitTriple(Triple triple) throws GenyrisException;

	public void visitGraph(Graph store) throws GenyrisException;

	public void visitDynamicSymbol(DynamicSymbol symbol)  throws GenyrisException;

	public void visitJavaWrapper(JavaWrapper javaWrapper)  throws GenyrisException;

	public void visitPipe(Pipe pipe)  throws GenyrisException;

    public void visitTailCall(TailCall tc) throws GenyrisException;

}

File: src/org/genyris/core/Visitor.kt
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.core

import org.genyris.dl.AbstractGraph
import org.genyris.dl.Triple
import org.genyris.exception.GenyrisException
import org.genyris.interp.EagerProcedure
import org.genyris.interp.LazyProcedure
import org.genyris.interp.TailCall
import org.genyris.io.pipe.Pipe
import org.genyris.java.JavaWrapper

interface Visitor {
    @kotlin.Throws(GenyrisException::class)
    fun visitPair(cons: Pair?)

    @kotlin.Throws(GenyrisException::class)
    fun visitStrinG(lst: StrinG?)

    @kotlin.Throws(GenyrisException::class)
    fun visitLazyProc(lproc: LazyProcedure?)

    @kotlin.Throws(GenyrisException::class)
    fun visitEagerProc(eproc: EagerProcedure?)

    @kotlin.Throws(GenyrisException::class)
    fun visitDictionary(frame: Dictionary?)

    @kotlin.Throws(GenyrisException::class)
    fun visitBignum(bignum: Bignum?)

    @kotlin.Throws(GenyrisException::class)
    fun visitStandardClass(klass: StandardClass?)

    @kotlin.Throws(GenyrisException::class)
    fun visitExpWithEmbeddedClasses(exp: ExpWithEmbeddedClasses?)

    @kotlin.Throws(GenyrisException::class)
    fun visitSimpleSymbol(simpleSymbol: SimpleSymbol?)

    @kotlin.Throws(GenyrisException::class)
    fun visitFullyQualifiedSymbol(sym: EscapedSymbol?)

    @kotlin.Throws(GenyrisException::class)
    fun visitFullyQualifiedSymbol(sym: URISymbol?)

    @kotlin.Throws(GenyrisException::class)
    fun visitPrefixSymbol(sym: PrefixSymbol?)

    @kotlin.Throws(GenyrisException::class)
    fun visitDynamicSymbol(symbol: DynamicSymbol?)

    @kotlin.Throws(GenyrisException::class)
    fun visitTriple(triple: Triple?)

    @kotlin.Throws(GenyrisException::class)
    fun visitGraph(store: AbstractGraph?)

    @kotlin.Throws(GenyrisException::class)
    fun visitJavaWrapper(javaWrapper: JavaWrapper?)

    @kotlin.Throws(GenyrisException::class)
    fun visitPipe(pipe: Pipe?)

    @kotlin.Throws(GenyrisException::class)
    fun visitTailCall(tc: TailCall?)

    @kotlin.Throws(GenyrisException::class)
    fun visitBiscuit(biscuit: Biscuit?)
}
